aicsimageio/writers/ome_zarr_writer.py

```diff
@@ -249,26 +265,39 @@ def write_image(
             * image_data.shape[ydimindex]
             * image_data.itemsize
         )
+
         target_chunk_size = 16 * (1024 * 1024)  # 16 MB
         # this is making an assumption of chunking whole XY planes.
         # TODO allow callers to configure chunk dims?
         nplanes_per_chunk = int(math.ceil(target_chunk_size / plane_size))
         nplanes_per_chunk = (
             min(nplanes_per_chunk, image_data.shape[zdimindex]) if zdimindex > -1 else 1
         )
-        chunk_dim_map = {
-            "T": 1,
-            "C": 1,
-            "Z": nplanes_per_chunk,
-            "Y": image_data.shape[ydimindex],
-            "X": image_data.shape[xdimindex],
-        }
-        chunk_dims = [
+
+        if chunk_dims is None:
+            chunk_dim_map = {
+                DimensionNames.Time: 1,
+                DimensionNames.Channel: 1,
+                DimensionNames.SpatialZ: nplanes_per_chunk,
+                DimensionNames.SpatialY: image_data.shape[ydimindex],
+                DimensionNames.SpatialX: image_data.shape[xdimindex],
+            }
+            chunk_dims = OmeZarrWriter.build_chunk_dims(
+                chunk_dim_map=chunk_dim_map, dimension_order=dimension_order
+            )
+        else:
+            chunk_dim_map = {
+                DimensionNames.Time: 1,
+                DimensionNames.Channel: 1,
+            }
```

Comment:
I guess this is indicating that something is still wrong when it's time to calculate the downsampled chunk sizes -- because they are always using nplanes_per_chunk, even if preferred chunk dims were passed in.